### DISCUSSION TITLE
Can I run migration programatically?

### DISCUSSION THREAD

**Discussion by dhavalhirdhav** [adonisjs/core#1907] [Help] [answered]
I am developing an application to support multi tenancy. So far I have been able to make it work to create database and also insert records into the respective tenant database. However when creating database, it does not run migration on the newly created database. Is there a way to run migration programmatically on the database that we specify? or do we need to rely on node exec function / method?

**Comment by steschwa** [comment, marked as answer]
I didn't try it yet but take a look into the `migration:run` command: https://github.com/adonisjs/lucid/blob/develop/commands/Migration/Run.ts

The command executer calls `await Run.run()` which will setup the `Migrator` with application and connection details and finally calls `Migrator.run()` to migrate files from `db/migrations`.

You should be able to reverse engineer the required calls to run the migration files.

You may also consider just executing the `migration:run` command from within your code.

**Reply by dhavalhirdhav** [reply]
ya, I was thinking of using exec but thought if there is a better way why not use that.
